packages/sdk/src/adapters/multipeer/rules.ts

```diff
@@ -1047,6 +1047,15 @@ export const Rules: { [id in Payloads.PayloadType]: Rule } = {
 
     // ========================================================================
     'user-update': {
-        ...ClientOnlyRule
+        ...DefaultRule,
+        client: {
+            beforeQueueMessageForClient: (
+                session: Session,
+                client: Client,
+                message: Message<Payloads.UserUpdate>
+            ) => {
+                return client.userId === message.payload.user.id ? message : null;
```

Comment:
>return client.userId === message.payload.user.id ? message : null; [](start = 16, length = 66)

What are you trying to accomplish here? In this method I think you will need to do something similar to what 'actor-update' is doing above, and this kind of filtering will be done elsewhere.

Reply:
I want this message to only be sent to the client that is hosting the given user. The other clients don't know about other users AFAICT, so wouldn't be able to do anything with this info.

---
In reply to: [267581467](https://github.com/Microsoft/mixed-reality-extension-sdk/pull/270#discussion_r267581467) [](ancestors = 267581467)